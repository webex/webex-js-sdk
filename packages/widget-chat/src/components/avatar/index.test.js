import React from 'react';
import renderer from 'react-test-renderer';

import Avatar from '.';

function createAvatar({image, isSelfAvatar, name}) {
  return renderer.create(
    <Avatar image={image} isSelfAvatar={isSelfAvatar} name={name} />
  );
}

describe(`Avatar component`, () => {
  let props;

  beforeEach(() => {
    props = {
      name: `Test User`,
      image: ``,
      isSelfAvatar: false
    };
  });

  it(`renders properly without image`, () => {
    props.image = null;
    expect(createAvatar(props)).toMatchSnapshot();
  });

  it(`renders properly as self avatar`, () => {
    props.isSelfAvatar = true;
    expect(createAvatar(props)).toMatchSnapshot();
  });

  it(`errors without name`, () => {
    props.name = ``;
    expect(createAvatar(props)).toThrow();
  });
});
