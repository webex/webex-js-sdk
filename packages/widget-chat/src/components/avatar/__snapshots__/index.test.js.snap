exports[`Avatar component renders properly as self avatar 1`] = `
<div
  className="avatar"
  style={undefined}>
  <span
    className="avatar-self" />
</div>
`;

exports[`Avatar component renders properly without image 1`] = `
<div
  className="avatar"
  style={undefined}>
  <span
    className="avatar-letter">
    T
  </span>
</div>
`;
